app/assets/javascripts/govuk_publishing_components/lib/auto-scroll-tracker.js

```diff
@@ -0,0 +1,170 @@
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+
+(function (Modules) {
+  function AutoScrollTracker ($module) {
+    this.$module = $module
+    this.SCROLL_TIMEOUT_DELAY = 20
+    this.RESIZE_TIMEOUT_DELAY = 100
+    this.PAGEHEIGHT_TIMEOUT_DELAY = 500
+    this.pageHeight = document.querySelector('body').clientHeight
+    this.trackedNodes = []
+    this.config = {
+      ignoreHeadingsInside: ['.govuk-cookie-banner', '.gem-c-feedback', 'footer'],
```

Comment:
That might be simpler, yes. I did it this way because I thought maybe at some point we might want an option on the script that would allow other elements to be ignored as well (say if we wanted to ignore the headings in the sidebar on a particular page). But we're not doing that yet, so... I'll have a think. It's probably simpler the way you suggest.